code-studio/src/js/components/Grid.jsx

```diff
@@ -28,33 +45,40 @@ var Cell = React.createClass({
       classNames.push('highlighted');
     }
 
-    var tiles = ['border', 'path', 'start', 'end', 'obstacle'];
-    classNames.push(tiles[cell.tileType_]);
-
-    if (mazeUtils.isBeeSkin(this.props.skin)) {
-      var conditions = ['', 'flower-or-hive', 'flower-or-nothing', 'hive-or-nothing', 'flower-hive-or-nothing'];
-      var features = ['hive', 'flower'];
-      if (cell.isVariableCloud()) {
-        classNames.push('conditional');
-        classNames.push(conditions[cell.cloudType_]);
-      } else if (cell.featureType_ !== undefined) {
-        classNames.push(features[cell.featureType_]);
+    // Cell type-specific stuff
+    // TODO(elijah) This area contains a bunch of application-specific
+    // logic which would ideally live somewhere more flexible.
```

Comment:
Agree with the sentiment of the comment, that it would be nice to clean up some of the app-specific stuff (tho I don't see an immediately obvious way to do that)